File: 2588.cpp
#include <unordered_map>

struct Solution {
    long long beautifulSubarrays(vector<int>& nums) {
        long long res = 0;
        int prefix = 0;
        unordered_map<int, int> h;
        h[0] = 1;
        for (int x : nums){
            prefix ^= x;
            res += h[prefix]++;
        }
        return res;
    }
};

File: sol/2588/2588.py
from collections import defaultdict

class Solution:
    def beautifulSubarrays(self, nums: List[int]) -> int:
        res = 0
        h = defaultdict(int)
        h[0] = 1
        prefix = 0
        for x in nums:
            prefix ^= x
            res += h[prefix]
            h[prefix] += 1
        return res
